### PR TITLE
fix: Display keyhashes

### PR DESCRIPTION
**Pull request by stackchain**
## Important
There are some implementations required in `yoroi-lib` to improve what we display to the users. Right now the lib for mobile doesn't support `EnterpriseAddress` and `PointerAddress`. We are also not resolving the script key hashes.

### CODE REVIEW

**Comment by stackchain** [comment]
This implementation should be replaced by `yoroi-lib`.

File: src/Receive/AddressModal.tsx [hunk 1]
```diff
@@ -156,11 +156,15 @@ const useStrings = () => {
 }
 
 const useKeyHashes = (address) => {
-  const [keyHashes, setKeyHashes] = React.useState<null | {staking: string; spending: string}>(null)
+  const [keyHashes, setKeyHashes] = React.useState<{staking: string | null; spending: string | null}>({
```
Comment (wolverineks) [inline src/Receive/AddressModal.tsx]: useState<CardanoKeyHashes>

File: src/yoroi-wallets/cardano/addressInfo.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  return keyHash.toBytes().then((bytes) => Buffer.from(bytes ?? '').toString('hex'))
+}
+
+/**
+ * @description Try to resolve the spending and staking key hashes for a bech32 address
+ *
+ * @param {string} address expects to be a bech32
+ * @returns {Promise<CardanoKeyHashes>}
+ */
+export async function getKeyHashes(address: string): Promise<CardanoKeyHashes> {
```
Comment (wolverineks) [inline src/yoroi-wallets/cardano/addressInfo.ts]: should these be split up and loaded separately?

File: src/Receive/AddressModal.tsx [hunk 3]
```diff
@@ -156,11 +156,15 @@ const useStrings = () => {
...
+    staking: null,
+    spending: null,
+  })
 
   React.useEffect(() => {
-    const addressInfo = new AddressDTOCardano(address)
-    addressInfo.getKeyHashes().then(setKeyHashes)
+    getKeyHashes(address)
+      .then(setKeyHashes)
+      .catch((e) => console.error(e))
```
Comment (banklesss) [inline src/Receive/AddressModal.tsx]:
```suggestion
      .catch(console.error)
```

File: src/yoroi-wallets/cardano/addressInfo.ts (hunk 2 above)
Comment (stackchain) [reply src/yoroi-wallets/cardano/addressInfo.ts]: it does

**Review by banklesss — APPROVED** [review]

File: src/yoroi-wallets/cardano/addressInfo.ts (hunk 2 above)
Comment (wolverineks) [reply src/yoroi-wallets/cardano/addressInfo.ts]: i meant, 2 different hooks, then you can move them into separate components with separate boundaries

Comment (stackchain) [reply src/yoroi-wallets/cardano/addressInfo.ts]: I think we will always display them together, we are using it only in 1 place, but now I get your point, will do one more iteration, then when lib comes out with the hooks will be ready.

**Pull request merged**